client/src/Components/Layout/Navbar/CardInputSearch.jsx

```diff
@@ -0,0 +1,88 @@
+import { Divider } from "@mui/material";
+import axios from "axios";
+import React, { useEffect, useRef, useState } from "react";
+import { Link } from "react-router-dom";
+
+export default function CardInputSearch() {
+  const [searchGyms, setSearchGyms] = useState([]);
+  const [isShowResult, setIsShowResult] = useState(false);
+  const [search, setSearch] = useState("");
+
+  const ref = useRef();
+
+  const handleBlur = () => {
+    setIsShowResult(false);
+    setSearch("");
+  };
+
+  // function to handle Click Outside input search box
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (ref.current && !ref.current.contains(event.target)) {
+        handleBlur();
+      }
+    };
+    document.addEventListener("click", handleClickOutside);
+    return () => {
+      document.removeEventListener("click", handleClickOutside);
+    };
+  }, []);
+
+  // function to handle Search and send to API
+  const handleSearch = async (event) => {
+    setSearch(event.target.value);
+    setIsShowResult(true);
+
+    // send request to server then set searchGyms to the response
+    const { data } = await axios({
```

Comment:
change `handleSearch` only to `setSearch`,
then in separate `useEffect` send request whenever search value changes